### PR TITLE
continuous deployment

### LINKED ISSUE #1387: Continuous Deployment of deployment examples
# Description
Currently we have ocis.owncloud.works as a permanent testing instance which always runs the latest master version of oCIS. It also serves as "demo" instance, in the sense, that we show what we got so far (but not in the sense of a reliable marketing instance).

This deployment reflects only one of our deployment examples (see here: https://owncloud.github.io/ocis/deployment/). So in the future all deployment examples should be deployed.

Beginning with Sprint 26 we release after every sprint. Therefore it makes sense to have a "last-release" instances besides the "latest" / "current release state" instances.

This should reflect in URLs:

ocis.basic.latest.owncloud.works
ocis.basic.released.owncloud.works
ocis.keycloak.latest.owncloud.works
ocis.keycloak.released.owncloud.works
...

Since there are a lot of long URLs, there must be an overview in docs about what instances exist.

Also all of these oCIS instances should be set up in a way to report traces and metrics to our monitoring and tracing server prometheus/jaeger/grafana.infra.owncloud.works

The file storage of the oCIS instances must be regulary wiped.

# User Stories
- As developer I want to have oCIS instances at hand where I can do quick testing of the lastest oCIS development state, so that I don't have to start it on my own
- As developer I want to have oCIS instances at hand where I can do quick testing of the last release of oCIS development state, so that I don't have to start it on my own
- As a client developer I want to have a quick way to test clients against oCIS with different IDPs (eg. default and Keycloak), so that I can test if it is working. Therefore I need an oCIS with public resolvable DNS and a certificate trusted by common OS's and browsers.
- As interested person in oCIS I want to have a simple way to try oCIS, since I got no time / knowledge to set it up on my own. (For this reason we also have https://ocis.owncloud.com/ but this is a stable deplyoment which only reflects one deployment scenario)
- As a legal representative of ownCloud I want files to be regularly be pruned on publicly accessible instances, so that these are not used to host bad stuff. (from here arises the whish of an private oCIS instance #OCIS-1462)

# Acceptance Criteria
- Instances are set up
- Instances are properly and continously updated

### PR DESCRIPTION
**Pull request by wkloucek**
## Description
Add continuous deployment of our deployment examples to CI.

## Related Issue
- Fixes #1387
- Blocked by #1174 9that PR is merged now, so not blocked any more)

## Motivation and Context
Having up to date test instances.

## How Has This Been Tested?
- in CI https://drone.owncloud.com/owncloud/ocis/3059/1/3

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised:

### CODE REVIEW

File: deployments/continuous-deployment-config/cs3_users_ocis/latest.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+- name: continuous-deployment-ocis-cs3-users-latest
+  server:
+    server_type: cx21
+    image: ubuntu-20.04
+    location: nbg1
+    inital_ssh_key_names:
```
Comment (phil-davis) [inline deployments/continuous-deployment-config/cs3_users_ocis/latest.yml:7]:
```suggestion
    initial_ssh_key_names:
```

File: deployments/continuous-deployment-config/cs3_users_ocis/released.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+---
+- name: continuous-deployment-ocis-cs3-users-released
+  server:
+    server_type: cx21
+    image: ubuntu-20.04
+    location: nbg1
+    inital_ssh_key_names:
```
Comment (phil-davis) [inline deployments/continuous-deployment-config/cs3_users_ocis/released.yml:7]:
```suggestion
    initial_ssh_key_names:
```

File: deployments/continuous-deployment-config/ocis_keycloak/latest.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+---
+- name: continuous-deployment-ocis-keycloak-latest
+  server:
+    server_type: cx21
+    image: ubuntu-20.04
+    location: nbg1
+    inital_ssh_key_names:
```
Comment (phil-davis) [inline deployments/continuous-deployment-config/ocis_keycloak/latest.yml:7]:
```suggestion
    initial_ssh_key_names:
```

File: deployments/continuous-deployment-config/ocis_keycloak/released.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+---
+- name: continuous-deployment-ocis-keycloak-released
+  server:
+    server_type: cx21
+    image: ubuntu-20.04
+    location: nbg1
+    inital_ssh_key_names:
```
Comment (phil-davis) [inline deployments/continuous-deployment-config/ocis_keycloak/released.yml:7]:
```suggestion
    initial_ssh_key_names:
```

File: deployments/continuous-deployment-config/ocis_traefik/latest.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+---
+- name: continuous-deployment-ocis-traefik-latest
+  server:
+    server_type: cx21
+    image: ubuntu-20.04
+    location: nbg1
+    inital_ssh_key_names:
```
Comment (phil-davis) [inline deployments/continuous-deployment-config/ocis_traefik/latest.yml:7]:
```suggestion
    initial_ssh_key_names:
```

File: deployments/continuous-deployment-config/ocis_traefik/released.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
+---
+- name: continuous-deployment-ocis-traefik-released
+  server:
+    server_type: cx21
+    image: ubuntu-20.04
+    location: nbg1
+    inital_ssh_key_names:
```
Comment (phil-davis) [inline deployments/continuous-deployment-config/ocis_traefik/released.yml:7]:
```suggestion
    initial_ssh_key_names:
```

**Review by phil-davis — COMMENTED** [review]
It would be good to fix wherever `inital_ssh_key_names` is used to be `initial_ssh_key_names`

File: deployments/continuous-deployment-config/ocis_traefik/released.yml (hunk 6 above)
Comment (wkloucek) [reply deployments/continuous-deployment-config/ocis_traefik/released.yml]: also fixed in https://github.com/owncloud-devops/continuous-deployment/commit/b35286d1f8c2fc40d14ceef74580a24fa70ebb70

**Review by phil-davis — APPROVED** [review]

**Pull request merged**